public/app/plugins/panel/geomap/layers/data/photosLayer.tsx

```diff
@@ -0,0 +1,212 @@
+  shadow: true,
+  crop: true,
+  radius: 20,
+  color: 'rgb(200, 200, 200)',
+};
+
+export const PHOTOS_LAYER_ID = 'photos';
+
+// Used by default when nothing is configured
+export const defaultPhotosConfig: MapLayerOptions<PhotoConfig> = {
+  type: PHOTOS_LAYER_ID,
+  name: '', // will get replaced
+  config: defaultOptions,
+  location: {
+    mode: FrameGeometrySourceMode.Auto,
+  },
+  tooltip: true,
+};
+
+// TODO Find a way to use SVG scaled to behave like a png, currently using base64 conversion
+//const unknownImage = '../../../../../public/img/icons/unicons/question-circle.svg';
+const unknownImage =
+  'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI0IiBjbGFzcz0iY3NzLWV5eDRkbyI+PHBhdGggZD0iTTExLjI5LDE1LjI5YTEuNTgsMS41OCwwLDAsMC0uMTIuMTUuNzYuNzYsMCwwLDAtLjA5LjE4LjY0LjY0LDAsMCwwLS4wNi4xOCwxLjM2LDEuMzYsMCwwLDAsMCwuMi44NC44NCwwLDAsMCwuMDguMzguOS45LDAsMCwwLC41NC41NC45NC45NCwwLDAsMCwuNzYsMCwuOS45LDAsMCwwLC41NC0uNTRBMSwxLDAsMCwwLDEzLDE2YTEsMSwwLDAsMC0uMjktLjcxQTEsMSwwLDAsMCwxMS4yOSwxNS4yOVpNMTIsMkExMCwxMCwwLDEsMCwyMiwxMiwxMCwxMCwwLDAsMCwxMiwyWm0wLDE4YTgsOCwwLDEsMSw4LThBOCw4LDAsMCwxLDEyLDIwWk0xMiw3QTMsMywwLDAsMCw5LjQsOC41YTEsMSwwLDEsMCwxLjczLDFBMSwxLDAsMCwxLDEyLDlhMSwxLDAsMCwxLDAsMiwxLDEsMCwwLDAtMSwxdjFhMSwxLDAsMCwwLDIsMHYtLjE4QTMsMywwLDAsMCwxMiw3WiI+PC9wYXRoPjwvc3ZnPgo=';
+
+/**
+ * Map layer configuration for circle overlay
+ */
+export const photosLayer: MapLayerRegistryItem<PhotoConfig> = {
+  id: PHOTOS_LAYER_ID,
+  name: 'Photos',
+  description: 'Render photos at each data point',
+  isBaseMap: false,
+  showLocation: true,
+  hideOpacity: true,
+  state: PluginState.alpha,
+
+  /**
+   * Function that configures transformation and returns a transformer
+   * @param map
+   * @param options
+   * @param theme
+   */
+  create: async (map: Map, options: MapLayerOptions<PhotoConfig>, eventBus: EventBus, theme: GrafanaTheme2) => {
+    // Assert default values
+    const config = {
+      ...defaultOptions,
+      ...options?.config,
+    };
+
+    const location = await getLocationMatchers(options.location);
+    const source = new FrameVectorSource(location);
+    const vectorLayer = new VectorLayer({
+      source,
+    });
+
+    let images: string[] = [];
+
+    vectorLayer.setStyle((feature: FeatureLike) => {
+      let src = unknownImage;
+      if (images.length > 0) {
+        const idx = feature.get('rowIndex') as number;
+        src = images[idx] ?? unknownImage;
+      }
+      return new Style({
+        image: new Photo({
+          src,
+          radius: config.radius,
+          crop: config.crop,
+          kind: config.kind,
+          shadow: config.shadow,
+          stroke: new Stroke({
+            width: config.border ?? 0,
+            color: theme.visualization.getColorByName(config.color),
+          }),
+          onload: () => {
+            vectorLayer.changed(); // ensure vector layer is rendered properly after image load
```

Comment:
I'm still seeing a few inconsistencies with image loading, but no where near as glitchy as it seemed before. Mainly a few images dont seem to render for a few seconds despite being on screen


https://user-images.githubusercontent.com/22381771/198146923-e102080b-a477-4687-963e-95a0f8101dbb.mov